examples/js/lines/LineSegments2.js

```diff
@@ -54,6 +54,56 @@ THREE.LineSegments2.prototype = Object.assign( Object.create( THREE.Mesh.prototy
 
 	}() ),
 
+	raycast: ( function () {
+
+		var ray = new THREE.Ray();
+		var inverseMatrix = new THREE.Matrix4();
+		var pointOnSegment = new THREE.Vector3();
+		var pointOnRay = new THREE.Vector3();
+		var startPoint3 = new THREE.Vector3();
+		var endPoint3 = new THREE.Vector3();
+
+		return function raycast( raycaster, intersects ) {
+
+			var precision = raycaster.linePrecision; // todo: calculate proper precision
```

Comment:
Line width is in pixels. What you are doing here is not correct. You can't mix world units and pixels. Maybe, if you see how sprite raycasting is handled, there is a workaround.

Reply:
Well, that won't work because lines can be fat and are specified in pixels -- unless you are willing to require to user to set the precision based on the camera settings and screen resolution, and accept that approach as "good-enough".

Reply:
@WestLangley , @Mugen87 , yes you rigth. In our proj we have another implementation: we project 3d points into 2d pixel coords using `Vector3.project(camera)` so then it is ok to compare cursor coordinates and linewidth, but also we use our custom raycaster contains camera.. I checked raycast method for sprites, am i right it is possible to project point3 onto a view plane using `this.modelViewMatrix` (seems function `transformVertex` in src/objects/Spite.js does what i described)? If yes our approach can be easily reimplemented without using `camera`.

Reply:
@weeklyTea I am glad fat lines are working for you! Perhaps you will be able to rework this PR. Also see #14936. Some of the issues may be similar...

Reply:
Yes, I mentioned option (2), but I don't like it, either. If you want to file a PR for (1), we can have a look. I do not know if (1) will be acceptable to others or not.